docs/testing/integration-tests.md

```diff
@@ -0,0 +1,200 @@
+    Cargo.toml        ⟵ Cargo.toml file specifying run configuration
+```
+
+Notice we have 2 `Cargo.toml` files in this project. The first is the main `Cargo.toml` file which is used to specify the dependencies of the project, and the second which is used to specify the dependencies of the integration tests. A sample configuration for the integration tests is shown below:
+
+```toml
+[package]
+name = "non-fungible-token-integration-tests"
+version = "1.0.0"
+publish = false
+edition = "2018"
+
+[dev-dependencies]
+near-sdk = "4.0.0"
+anyhow = "1.0"
+borsh = "0.9"
+maplit = "1.0"
+near-units = "0.2.0"
+# arbitrary_precision enabled for u128 types that workspaces requires for Balance types
+serde_json = { version = "1.0", features = ["arbitrary_precision"] }
+tokio = { version = "1.18.1", features = ["full"] }
+tracing = "0.1"
+tracing-subscriber = { version = "0.3.11", features = ["env-filter"] }
+workspaces = "0.3.1"
+pkg-config = "0.3.1"
+
+[[example]]
+name = "integration-tests"
+path = "src/tests.rs"
+```
+
+The `tests.rs` file above will contain the integration tests, which will preserve state after each run if they are run in one file as in this setup. These can be run with the following command from the same level as the test `Cargo.toml` file:
+
+    cargo run --example integration-tests
+
+:::note
+In case you wish for a new state for a test run, you can create an additional `.rs` file in `tests/src` which you can reference as an additional `[[example]]` in the `Cargo.toml` file. You would similarly run these tests with the following command:
+
+    cargo run --example <name of the example>
+:::
+
+## Comparing an Example
+
+### Unit Test
+
+Let's take a look at a very simple unit test and integration test that accomplish the same thing. Normally you wouldn't duplicate efforts like this (as integration tests are intended to be broader in scope), but it will be informative.
+
+One of the simple examples on the <a href="https://near.dev" target="_blank">NEAR Examples landing page</a> is the <a href="https://examples.near.org/rust-counter" target="_blank">Rust counter</a>. We'll be using snippets from this repository to demonstrate simulation tests.
+
+First, note this unit test that tests the functionality of the `increment` method:
+
+```rust reference
+https://github.com/near-examples/rust-counter/blob/dad0cfd918fcc4c25611307aa07ad377b97ea52b/contract/src/lib.rs#L128-L139
+```
+
+The test above sets up the testing context, instantiates the smart contract's struct called `Counter`, calls the `increment` method, and makes an assertion that a number field on the struct is now `1`.
+
+Let's look at how this might be written with workspaces tests. The snippet below is a bit longer as it demonstrates a couple of things worth noting.
+
+### Workspaces Test
+
+```rust reference
+https://github.com/near-examples/rust-counter/blob/6a7af5a32c630e0298c09c24eab87267746552b2/integration-tests/rs/src/tests.rs#L6-L58
+```
```

Comment:
Just noting: wasm files is being fetched from an `out` directory in this example which does not correlate with `target` directory mentioned above which might be confusing. Perhaps it would be a good idea to explicitly say where the wasm should be pulled from.

Reply:
I added an explanation in the info box below. In the context of Rust, I believe naming the build out directory `target` might be confusing since `/target` is the dir name created when `cargo build`  is run in the project. I thought in the schematic above that `/target` was in fact that directory.

Reply:
I was referring to this specific line which pull wasm from `out`: https://github.com/near-examples/rust-counter/blob/6a7af5a32c630e0298c09c24eab87267746552b2/integration-tests/rs/src/tests.rs#L6

Not sure where this directory is coming from, probably some external script

Reply:
Have replaced example for FT, so this snippet is not in there anymore